sys/net/gnrc/sock/tcp/gnrc_sock_tcp.c

```diff
@@ -0,0 +1,146 @@
+/*
+ * Copyright (C) 2021 Simon Brummer <simon.brummer@posteo.de>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @{
+ *
+ * @file
+ * @brief GNRC implementation of @ref net_sock_tcp
+ *
+ * @author Simon Brummer <simon.brummer@posteo.de>
+ */
+
+#include <assert.h>
+
+#include "net/gnrc/tcp.h"
+#include "net/sock/tcp.h"
+#include "sock_types.h"
+
+int sock_tcp_connect(sock_tcp_t *sock, const sock_tcp_ep_t *remote,
+                     uint16_t local_port, uint16_t flags)
+{
+    /* Asserts defined by API. */
+    assert(sock != NULL);
+    assert(remote != NULL);
+    assert(remote->port != 0);
+
+    /* Asserts to protect GNRC_TCP. Flags are not supported. */
+    assert(flags == 0);
```

Comment:
```suggestion
    assert(flags == 0);
    (void)flags;
```

otherwise `flags` will be unused when build without `DEVELHELP=1`